src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -2398,6 +2414,9 @@ private void OnDebuggerStop(object sender, DebuggerStopEventArgs e)
                         this.PromptNest,
                         this.CurrentRunspace.Runspace))
                     {
+                        // Since we are no longer at a breakpoint, we set this to null.
+                        CurrentDebuggerStopEventArgs = null;
```

Comment:
Should save and restore instead?  I *think* it'll be possible for this to nest.  Same with the other one above.

Also should this use `Interlocked.Exchange`? (maybe in the getter as well?)

Reply:
So, I can't use `Interlocked.Exchange` because `CurrentDebuggerStopEventArgs` is a property. I could use a lock but honestly, I don't see why this would happen. If someone opens an issue, we can revisit.